backend/app/usecase/search/search.go

```diff
@@ -66,13 +69,118 @@ func (s Search) searchResource(resource Resource, orderBy order.Order, query Que
 }
 
 func (s Search) searchShortLink(query Query, orderBy order.Order, filter Filter) (Result, error) {
-	return Result{}, nil
+	if query.User == nil {
+		return Result{}, errors.New("user not provided")
+	}
+	if len(query.Query) == 0 {
+		return Result{}, errors.New("query not provided")
+	}
+
+	shortLinks, err := s.getShortLinkByUser(*query.User)
+	if err != nil {
+		return Result{}, err
+	}
+
+	matchedShortLinks := matchShortLinks(shortLinks, query, orderBy)
+
+	filteredShortLinks := filterShortLinks(matchedShortLinks, filter)
+
+	return Result{
+		shortLinks: filteredShortLinks,
+		users:      nil,
+	}, nil
 }
 
 func (s Search) searchUser(query Query, orderBy order.Order, filter Filter) (Result, error) {
 	return Result{}, nil
 }
 
+func (s Search) getShortLinkByUser(user entity.User) ([]entity.ShortLink, error) {
+	aliases, err := s.userShortLinkRepo.FindAliasesByUser(user)
+	if err != nil {
+		return []entity.ShortLink{}, err
+	}
+
+	return s.shortLinkRepo.GetShortLinksByAliases(aliases)
+}
+
+func matchShortLinks(shortLinks []entity.ShortLink, query Query, orderBy order.Order) []entity.ShortLink {
```

Comment:
Can we eliminate comments by adding an extra layer of methods? Feel free to disagree.
Another advantage is that, if `aliasByAnd` is true, there is no need to calculate for `aliasByOr`
For example,
```
func matchShortLinks(shortLinks []entity.ShortLink, query Query, orderBy order.Order) []entity.ShortLink {
	keywords := getKeywordsFromQuery(query.Query)
	
	for _, shortLink := range shortLinks {
		if aliasContainsAllKeywords(shortLink.Alias, keywords) {

		} else if aliasContainsAnyKeyword(shortLink.Alias, keywords) {

		} else if longLinkContainsAllKeywords(shortLink.LongLink, keywords) {

		} else if longLinkContainsAnyKeyword(shortLink.LongLink, keywords) {

		}
	}
}

func aliasContainsAllKeywords(alias string, keywords []string) {
	return stringContainsAll(alias, keywords);
}

func aliasContainsAnyKeyword(alias string, keywords []string) {
	return stringContainsAny(alias, keywords);
}

func stringContainsAll(str string, patterns []string) {
	// logic
}
```

Reply:
I do agree at some level! I think eliminating comments by adding an extra layer of methods is the right thing for readability and optimization, but I wouldn't differentiate function aliasContainsAllKeywords from longLinkContainsAllKeywords since they are the same and that's why I kept those functions under containsAllWords. Please checkout the modification after your suggestion, what do you think?

Reply:
Agreed